dora/core/server/worker/src/main/java/alluxio/worker/http/HttpServerHandler.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.http;
+
+import static io.netty.handler.codec.http.HttpHeaderNames.CONNECTION;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_LENGTH;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE;
+import static io.netty.handler.codec.http.HttpHeaderValues.CLOSE;
+import static io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
+import static io.netty.handler.codec.http.HttpHeaderValues.TEXT_PLAIN;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * {@link HttpServerHandler} deals with HTTP requests received from Netty Channel.
+ */
+public class HttpServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+
+  private final PagedService mPagedService;
+
+  /**
+   * {@link HttpServerHandler} deals with HTTP requests received from Netty Channel.
+   * @param pagedService the {@link PagedService} object provides page related RESTful API
+   */
+  public HttpServerHandler(PagedService pagedService) {
+    mPagedService = pagedService;
+  }
+
+  @Override
+  public void channelReadComplete(ChannelHandlerContext ctx) {
+    ctx.flush();
+  }
+
+  @Override
+  public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) {
+    if (msg instanceof HttpRequest) {
+      HttpRequest req = (HttpRequest) msg;
+      String requestUri = req.uri();
+      // parse the request uri to get the parameters
+      // TODO(JiamingMai): parse the URI and dispatch it to different methods
+      requestUri = requestUri.substring(requestUri.indexOf("?") + 1);
+      String[] params = requestUri.split("&");
```

Comment:
Just a quick question: For the current testing with the AI training use case (computer vision), the URI is something like `http://localhost:28080/page?fileId=xxx&pageIndex=xxx`. Could you elaborate a bit more on how to figure out the `pageIndex` on the python client side?

Reply:
If the size of image file is less than 1MB (default page size), just let pageIndex=0.